Detectors/Base/include/DetectorsBase/ContVec.h

```diff
@@ -0,0 +1,362 @@
+#ifndef O2_BASE_CONTVEC_H
+#define O2_BASE_CONTVEC_H
+
+#include <TTree.h>
+#include <TBranch.h>
+#include <TBranchElement.h>
+#include <TBuffer.h>
+#include <TObject.h>
+#include <memory>
+#include <algorithm>
+#include <type_traits>
+#include <stdexcept>
+#include <iostream>
+#include <FairLogger.h>
+
+
+namespace o2 {
+namespace Base {
+
+using sizeType = int;
+
+ template <class T, class H>
+  class ContVec: public TObject {
+
+ public:
+  
+  struct Header {
+    H         userInfo;         // user assigned data info
+    int       expandPolicy;     // user assigned policy: n>0 -> new=old+n, n<=0 -> new=2*(old+n)
+    sizeType  sizeInBytes;      // total booked size
+    sizeType  nObjects;         // number of objects stored
+  };
+  // main constructor
+  ContVec(sizeType iniSize=0, int expPol=-100);
+
+  // construct from received raw pointer on the existing buffer, see recreate comments
+  ContVec(char* rawptr, bool copy, sizeType nbytes=-1)  : mPtr(nullptr) {recreate(rawptr,copy,nbytes);}
+
+  // recreate container from received raw pointer on the existing buffer
+  void  recreate(char* rawptr, bool copy, sizeType nbytes=-1);
```

Comment:
I think a better name would be "adopt" - since you would be adopting an existing buffer. Another issue is the ownership - you take ownership of a raw pointer here for a resource which might be (and usually is) owned by the transport. If you want to transfer ownership take a unique_ptr as argument. We should think about what to do here, maybe it would be better to take the message (FairMQMessage) directly which is a movable owning thing or come up with a different general API/agreement on how to do buffer ownership transfers in and (especially) out of transport.